server/data/s3Client.ts

```diff
@@ -0,0 +1,57 @@
+import { SelectObjectContentCommandInput, S3, GetObjectCommand, SelectObjectContentCommand } from '@aws-sdk/client-s3'
+import { Readable } from 'stream'
+import getStream from 'get-stream'
+
+export interface S3BucketConfig {
+  accessKeyId: string
+  secretAccessKey: string
+  bucket: string
+  endpoint?: string
+}
+
+export default class S3Client {
+  s3: S3
+
+  bucket: string
+
+  constructor(config: S3BucketConfig) {
+    this.s3 = new S3({
+      apiVersion: '2006-03-01',
+      region: 'eu-west-2',
+      credentials: {
+        accessKeyId: config.accessKeyId,
+        secretAccessKey: config.secretAccessKey,
+      },
+      endpoint: config.endpoint,
+      forcePathStyle: true,
+    })
+    this.bucket = config.bucket
+  }
+
+  async getObject(key: string): Promise<string> {
+    const command = new GetObjectCommand({
+      Bucket: this.bucket,
+      Key: key,
+    })
+
+    const response = await this.s3.send(command)
+    return (await getStream.buffer(response.Body as Readable)).toString()
+  }
+
+  async selectObjectContent(s3selectParams: SelectObjectContentCommandInput): Promise<any[]> {
+    const command = new SelectObjectContentCommand(s3selectParams)
+    const response = await this.s3.send(command)
+
+    const records: Uint8Array[] = []
+    /* eslint-disable-next-line */
+    for await (const event of response.Payload) {
+      if (event.Records) {
+        records.push(event.Records.Payload)
+      }
+    }
```

Comment:
I tried something like this last week but the linter started to complain so I assumed it was a naughty thing to do, but this `for` loop versions reads/look so much better than those weird `on()` versions 👌

Reply:
My understanding is the eslint raises the issue because not all browsers support the Symbol.asyncIterator. We are using this server side so no browser support issues for this type of usage of `for ... of`